src/System.ComponentModel.TypeConverter/src/System/ComponentModel/BaseNumberConverter.cs

```diff
@@ -111,7 +111,7 @@ public override object ConvertTo(ITypeDescriptorContext context, CultureInfo cul
                 throw new ArgumentNullException(nameof(destinationType));
             }
 
-            if (destinationType == typeof(string) && value != null && TargetType.GetTypeInfo().IsAssignableFrom(value.GetType().GetTypeInfo()))
+            if (destinationType == typeof(string) && value != null && TargetType.IsInstanceOfType(value))
```

Comment:
Ah, nevermind, it looks like IsInstanceOfType just uses IsAssignableFrom internally.